course/layouts/learning_resource_types/list.html

```diff
@@ -0,0 +1,10 @@
+{{ define "main" }}
```

Comment:
FYI, this layout is actually not necessary if the desired functionality is to have the list page at `/resources` and in fact this generates an extra set of pages at `/learning_resource_types` that looks like this:

![image](https://user-images.githubusercontent.com/12089658/175098938-a5aff6b8-5a24-4e04-85c6-4704d9ff8270.png)

I neglected to realize and mention this before you merged https://github.com/mitodl/ocw-hugo-projects/pull/197, but we might want to disable XML output for the `learning_resource_types` taxonomy as well.  The setting you added in that PR will generate "pages" at `site.Pages` for the `learning_resource_types` taxonomy and iterating them manually in the list template at `course/layouts/resources/list.html` will still work.  I tested this locally by removing this layout and regenerating my site.  We also need to remove the default `categories` and `tags` taxonomies from the Hugo config, so I can include removing the XML output for `learning_resource_types` in that PR, but for now we don't need a list template here as it will generate extra HTML.  We have to be careful about unintentionally generating pages now because all of them are inserted into the sitemap and subsequently indexed by Google.

Reply:
Yea, I'll create an issue in `ocw-hugo-projects` to deal with the extra XML files being generated by the taxonomies.  In the meantime, @MAbdurrehman12 , like @Ferdi mentions below we should actually leave the `course/layouts/learning_resource_types/list.html` template because I misunderstood how you were using it and obviously it's central to how this whole thing functions.  Without it, the list pages for each learning resource type won't be generated.  However, we should clean up the title on that page so that it doesn't have underscores.  Instead of calling `course_content.html` you might want to create a new partial, something similar to `content_header.html` (maybe `taxonomy_header.html`?) that takes titles automatically inferred from types like that and strips out the underscores and title cases everything so it looks nice.

As far as the resulting base URL of the taxonomy lists (`/learning_resource_types`) I think we can use the `permalinks` section of the Hugo config for this as seen here: https://github.com/gohugoio/hugo/issues/1208#issuecomment-859222720.  I was able to get the taxonomy lists to be generated at `/resources/exams` etc. by adding this to the Hugo config:

```
permalinks:
  learning_resource_types: "/resources/:slug/"
```

The `/learning_resource_types/index.html` page is still generated for some reason, but that's not a big deal as long as we can fix the title and remove the icons, as Ferdi mentioned.

Reply:
Thanks for the review guys.

@gumaerc Now we have to create a PR in `hugo-projects` to 
- disable XML for `learning_resource_type` taxonomy
- disable  Hugo default taxonomies (`categories` and `tags`)
- add the config to get the taxonomy lists generated at `/resources/*type*`.

Is this correct?
If yes, then I can go on create the PR.

In the meanwhile, I think you can review this PR again and if all is good then we can merge it to get it to RC, if there's any issue, let me know.

Thanks

Reply:
@MAbdurrehman12 :

The XML output from Hugo comes from the "RSS" output format built into Hugo.  This is already disabled for the `home`, `page` and `section` kinds.  If you look at the `output` section in either of the `config.yaml` files, you'll see that "RSS" is not included as an output format.  The default configuration for outputs includes "HTML" and "RSS," and when you redefine this if you don't explicitly specify those they aren't included.  So, I think we'd just need to modify the `outputs` section like so:
 
```
outputs:
  home:
    - coursedata
    - contentmap
    - HTML
  page:
    - coursedata
    - HTML
  section:
    - coursedata
    - HTML
  taxonomy:
    - HTML
  term:
    - HTML
```

I talked with @Ferdi about RSS support and it is desirable, but I think we want to be more deliberate about what kinds of RSS feeds we provide, i.e. a feed of new / updated courses or something.  The automatic RSS feeds generated by Hugo would just list every piece of content available, which isn't super useful.

In the same vein, it seems that the default taxonomies are actually already disabled.  See here for more info: https://gohugo.io/content-management/taxonomies/#example-removing-default-taxonomies.  By having this in the config:

```
taxonomies:
  learning_resource_type: learning_resource_types
```

... we are already telling Hugo that these are the only taxonomies it should be creating.  I verified this locally by looking at the `public` output folder when running a course.  The `tags` and `categories` folders are not created.  They still exist in the live site bucket, and this is because the folders are not being cleared out of old content with each build.  For `ocw-www`, we might want to add:

```
disableKinds:
- taxonomy
- term
```

...until we have taxonomies that we want generated by `ocw-www`.  As far as the `permalinks` section I mentioned above, we should go ahead and do that because it simply instruct Hugo that for the `learning_resource_types` taxonomy, it should generate pages at `/resources`.